### PR TITLE
UINV-195 Set exchange rate manually

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UINV-195

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/invoices/InvoiceForm/CurrentExchangeRate/CurrentExchangeRate.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  beforeEach(() => {
+    mutator = {
+      exchangeRate: {
+        GET: jest.fn(),
+      },
+    };
+  });
+
+  afterEach(cleanup);
```
Comment (NikitaSedyx) [inline src/invoices/InvoiceForm/CurrentExchangeRate/CurrentExchangeRate.test.js:37]: not required

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 2]
```diff
@@ -442,6 +442,8 @@ const InvoiceForm = ({
                           label={<FormattedMessage id="ui-invoice.invoice.currentExchangeRate" />}
                           exchangeFrom={filledCurrency}
                           exchangeTo={systemCurrency}
+                          setExchangeRateEnabled={setExchangeRateEnabled}
```
Comment (NikitaSedyx) [inline src/invoices/InvoiceForm/InvoiceForm.js:445]: don't think it's right approach

imo you should use context

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 3]
```diff
@@ -466,6 +468,9 @@ const InvoiceForm = ({
                           type="number"
                           readOnly={!isExchangeRateEnabled}
                           tooltipText={tooltipTextExchangeRate}
+                          required={isExchangeRateRequired}
+                          validate={isExchangeRateRequired ? validateRequired : undefined}
+                          key={isExchangeRateRequired ? 1 : 0}
```
Comment (NikitaSedyx) [inline src/invoices/InvoiceForm/InvoiceForm.js:473]: really required?

File: src/invoices/InvoiceForm/CurrentExchangeRate/CurrentExchangeRate.test.js (hunk 1 above)
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/CurrentExchangeRate/CurrentExchangeRate.test.js:37]: it's not required AFAIK https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-cleanup

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 3 above)
Comment (aliaksei-chumakou) [reply src/invoices/InvoiceForm/InvoiceForm.js:473]: @NikitaSedyx there is a buggy behavior in rff, if validation requirement is dynamic we have to put key so component is rerendered. https://final-form.org/docs/react-final-form/types/FieldProps#validate 

> ⚠️ IMPORTANT ⚠️ – By default, in order to allow inline fat-arrow validation functions, the field will not rerender if you change your validation function to an alternate function that has a different behavior. If you need your field to rerender with a new validation function, you will need to update another prop on the Field, such as key. See the following sandbox for an example:

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js:445]: but anyway I need to use setState to change some values from the Child component

Comment (NikitaSedyx) [reply src/invoices/InvoiceForm/InvoiceForm.js:445]: making parent component dependent from child is not the best idea (especially without redux)

why not to move exchange rate loading to another level?

**Review by aliaksei-chumakou — APPROVED** [review]

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 2 above)
Comment (aliaksei-chumakou) [reply src/invoices/InvoiceForm/InvoiceForm.js:445]: this makes sense, especially if you can say without specific flags that input is required or enabled.

**Pull request merged**